src/jit/codegenxarch.cpp

```diff
@@ -757,7 +757,7 @@ void CodeGen::genCodeForBinary(GenTree* treeNode)
     }
     // now we know there are 3 different operands so attempt to use LEA
     else if (oper == GT_ADD && !varTypeIsFloating(treeNode) && !treeNode->gtOverflowEx() // LEA does not set flags
-             && (op2->isContainedIntOrIImmed() || !op2->isContained()))
+             && (op2->isContainedIntOrIImmed() || !op2->isContained()) && !treeNode->gtSetFlags())
```

Comment:
It seems that not using `lea` here would result in generating two `add` instructions. It would seem that we trade a `test` for an `add`, not much of an improvement.